Action/SetPayUAction.php

```diff
@@ -102,14 +110,47 @@ public function execute($request)
                 $order['products'] = $model['products'];
             }
 
+            if (isset($model['payMethods'])) {
```

Comment:
I think if we should not move the $order setup to a separate method - execute starts to be dangerously long.